hibernate-core/src/main/java/org/hibernate/type/descriptor/converter/AttributeConverterTypeAdapter.java

```diff
@@ -143,4 +145,32 @@ private Object getConvertedValue(AttributeConverter<T, Object> converter, T valu
 			throw new PersistenceException( "Error attempting to apply AttributeConverter", re );
 		}
 	}
+
+	@Override
+	public Object replace(
+			Object original,
+			Object target,
+			SharedSessionContractImplementor session,
+			Object owner,
+			Map<Object, Object> copyCache,
+			ForeignKeyDirection foreignKeyDirection) {
+		return ForeignKeyDirection.FROM_PARENT == foreignKeyDirection
+				? domainJtd.getReplacement( (T) original, (T) target, session )
+				: target;
+	}
+
+	@Override
+	public Object replace(
+			Object original,
+			Object target,
+			SharedSessionContractImplementor session,
+			Object owner,
+			Map<Object, Object> copyCache) {
+		if ( original == null && target == null ) {
+			return null;
+		}
+
+		return domainJtd.getReplacement( (T) original, (T) target, session );
+	}
+
```

Comment:
Shouldn't we instead call the super constructor with `super( std, domainJtd );` instead of with `super( std, (JavaType<T>) relationalJtd );`?